src/hls-adapter.js

```diff
@@ -1203,6 +1217,36 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     }
   }
 
+  /**
+   * called when a level is loaded
+   * @private
+   * @param {any} e - the event object
+   * @param {any} data - the event data
+   * @returns {void}
+   */
+  _onLevelLoaded = (e: any, data: any) => {
+    const {
+      details: {endSN}
+    } = data;
+    if (this._lastLoadedFragSN === endSN) {
+      this._sameFragSNLoadedCount++;
+      HlsAdapter._logger.debug(`Same frag SN. Count is: ${this._sameFragSNLoadedCount}, Max is: ${this._config.network.maxStaleLevelReloads}`);
+      if (this._sameFragSNLoadedCount >= this._config.network.maxStaleLevelReloads) {
+        HlsAdapter._logger.error(`Same frag loading reached max count`);
+        const error = new Error(Error.Severity.CRITICAL, Error.Category.NETWORK, Error.Code.LIVE_MANIFEST_REFRESH_ERROR, {
```

Comment:
Don't we want for this case to seekToLiveEdge and try to play in case of live stream is out of sync?